staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cache_simple.go

```diff
@@ -17,22 +17,29 @@ limitations under the License.
 package cache
 
 import (
+	"context"
+	"runtime"
 	"time"
 
-	lrucache "k8s.io/apimachinery/pkg/util/cache"
+	utilcache "k8s.io/apimachinery/pkg/util/cache"
 	"k8s.io/apimachinery/pkg/util/clock"
 )
 
 type simpleCache struct {
-	lru *lrucache.LRUExpireCache
+	*utilcache.Expiring
 }
 
-func newSimpleCache(size int, clock clock.Clock) cache {
-	return &simpleCache{lru: lrucache.NewLRUExpireCacheWithClock(size, clock)}
+func newSimpleCache(clock clock.Clock) cache {
+	ctx, cancel := context.WithCancel(context.Background())
+	c := &simpleCache{utilcache.NewExpiringWithClock(ctx, clock)}
+	runtime.SetFinalizer(c, func(*simpleCache) {
```

Comment:
Does this reliably work? Do we do this anywhere else in the code base? Consider making the caller provide the context instead?

Reply:
I'm with @enj, let's not use finalizers for this.

Reply:
Ok, will do. I'll warn you now that the diff is going to be significant.

Reply:
Done in part. How far do you want me to plumb this? We run this authenticator in all apiservers and kubelet, so plumbing down to main is a lot of change.